jetty-http/src/main/java/org/eclipse/jetty/http/HttpGenerator.java

```diff
@@ -570,13 +571,18 @@ private void generateResponseLine(MetaData.Response response, ByteBuffer header)
     }
 
     private void generateHeaders(ByteBuffer header, ByteBuffer content, boolean last)
+    {
+        generateHeaders(header, content == null ? null : new ByteBuffer[]{content}, BufferUtil.length(content), last);
+    }
+
+    private void generateHeaders(ByteBuffer header, ByteBuffer[] content, long contentSize, boolean last)
```

Comment:
I'd very much prefer it if we could avoid these temporary arrays and instead only work with contentLength:

```suggestion
    {
        generateHeaders(header, BufferUtil.length(content), last);
    }

    private void generateHeaders(ByteBuffer header, long contentSize, boolean last)
```

I think the generator only every uses the contentLength and then does the occassional clear.  Is the clear really necessary and could it be done outside of the loop?

Reply:
I think I can accomplish the same logic only clearing the buffer externally. The changes will be a bit more extensive and some extra logic will need to be done everywhere the generator is used.